internal/form/question/service.go

```diff
@@ -73,16 +74,21 @@ func (s *Service) Delete(ctx context.Context, formID uuid.UUID, id uuid.UUID) er
 	defer span.End()
 	logger := logutil.WithContext(ctx, s.logger)
 
-	err := s.queries.Delete(ctx, DeleteParams{
+	rowsAffected, err := s.queries.Delete(ctx, DeleteParams{
 		FormID: formID,
 		ID:     id,
 	})
 	if err != nil {
 		err = databaseutil.WrapDBError(err, logger, "delete question")
 		span.RecordError(err)
+		return err
 	}
 
-	return err
+	if rowsAffected == 0 {
+		return internal.ErrQuestionNotFound
```

Comment:
In RESTful API design, the DELETE method should be idempotent.
`DELETE /users/123` doesn’t mean “delete an object named 123”; it means “ensure this resource no longer exists.” If the resource is already absent, the system’s final state is still “nonexistent,” meaning the request has been fulfilled.